functional_programming.py

```diff
@@ -0,0 +1,56 @@
+
+def sequential_map(*args):
+    lst = list(args[-1])
+    for arg in args[:-1]:
+        lst = list(map(arg, lst))
+    return lst
+
+
+# # Test for sequential_map
+# print(sequential_map(np.square, np.sqrt, lambda x: x ** 3, [1, 2, 3, 4, 5]))
+
+def consensus_filter(*args):
+    lst = list(args[-1])
+    for arg in args[:-1]:
+        lst = list(filter(arg, lst))
+    return lst
+
+
+# # Test for consensus_filter
+# print(consensus_filter(lambda x: x > 0, lambda x: x > 5, lambda x: x < 10, [-2, 0, 4, 6, 11]))
+
+
+def conditional_reduce(filter_function, reduce_function, lst):
+    filtered_lst = list(filter(filter_function, lst))
+    for i in range(len(filtered_lst) - 1):
+        filtered_lst[i + 1] = reduce_function(filtered_lst[i], filtered_lst[i + 1])
+    return filtered_lst[-1]
+
+
+# # Test for conditional_reduce
+# print(conditional_reduce(lambda x: x < 5, lambda x, y: x + y, [1, 3, 5, 10]))
+
+def func_chain(*args):
+    def returned_function(value):
+        value = [value]
+        for func in args:
+            value = list(map(func, value))
```

Comment:
В таком случае мы каждый раз будем получать на выходе список, что возможно не является для нас желаемым результатом. Лучше работать напрямую с входными значениями, в таком случае уже не понадобится map, а можно будет написать что-то в этом духе
```suggestion
        for func in args:
            value = func(value)
```